cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/aws-orchestrator/AwsOrchestrator.service.tsx

```diff
@@ -87,4 +89,50 @@ export class AwsOrchestrator {
             setErrorObject(addErrorInfo(errorObject, e, AWS_FIELDS.NODE_TYPE));
         }
     }
+
+    static initAvailabilityZones(props: AwsOrchestratorProps) {
+        const { awsDispatch, setErrorObject, errorObject } = props;
+        DefaultOrchestrator.initResources<AWSKeyPair>({
+            resourceName: AWS_FIELDS.AVAILABILITY_ZONES,
+            dispatch: awsDispatch,
+            errorObject,
+            setErrorObject,
+            fetcher: () => AwsService.getAwsAvailabilityZones(),
+        });
+    }
+
+    static async initNodeTypes4AZs(props: AwsOrchestratorProps, nodeProfile: string) {
+        const { awsState, awsDispatch, setErrorObject, errorObject } = props;
+        const azList = awsState[STORE_SECTION_RESOURCES][AWS_FIELDS.AVAILABILITY_ZONES];
+        const defaultAZNameList: string[] = AwsDefaults.defaulAvailabilityZoneNameStrategy(azList, nodeProfile);
+        const selectedAzs: { [key: string]: string }[] = [];
+
+        for (const az of defaultAZNameList) {
+            DefaultOrchestrator.initResources<{ [key: string]: string }>({
```

Comment:
I feel like there should be some kind of AwsNodeInstanceType structure to use in a call to `initResources<AwsNodeInstanceType[]>()`